src/fmeval/eval_algorithms/answer_relevance.py

```diff
@@ -0,0 +1,289 @@
+Question: When is the scheduled launch date and time for the PSLV-C56 mission, and where will it be launched from?
+<example>
+
+Now solve the following task.
+Answer:$answer
+Question:Assistant:"""  # noqa: E501
+
+
+@dataclass(frozen=True)
+class AnswerRelevanceConfig(EvalAlgorithmConfig):
+    """Configures the Answer Relevance evaluation algorithm.
+
+    :param strictness: The number questions generated per answer.
+    """
+
+    strictness: Optional[int] = 1
+
+
+class AnswerRelevanceScore(Transform):
+    """This transform augments its input record with the computed answer relevance score.
+
+    See the docstring for `Answer Relevance` for more details regarding the score itself.
+    """
+
+    def __init__(
+        self,
+        embeddings_model: ModelRunner,
+        model_output_key: str = DatasetColumns.MODEL_OUTPUT.value.name,
+        output_key: str = ANSWER_RELEVANCE,
+    ):
+        """AnswerRelevanceScore initializer.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param model_output_key: The key corresponding to the model output.
+        :param output_key: The key corresponding to the answer relevance score that will be added to the input record.
+        """
+        super().__init__(embeddings_model, model_output_key, output_key)
+        self.register_input_output_keys(
+            input_keys=["gen_questions", model_output_key],
+            output_keys=[output_key],
+        )
+        self.output_key = output_key
+        self.model_output_key = model_output_key
+        self.embeddings_model = embeddings_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed answer relevance score.
+
+        :param record: The input record.
+        :returns: The input record, with the answer relevance score added in.
+        """
+        gen_questions = record["gen_questions"].split("\n")
+        question = record[self.model_output_key]
+        record[self.output_key] = self._get_score(self.embeddings_model, question, gen_questions)
+        return record
+
+    @staticmethod
+    def _get_score(embeddings_model: ModelRunner, question: str, generated_questions: List[str]) -> float:
+        """Given generated questions and the original question, embedding model, computer Answer Relevance score.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param question: The given prompt (model input).
+        :param generated_questions: An artificial question generated by judge model based on the answer.
+        :returns: 0 to 1. See the docstring for `Answer Relevance` for more details
+            on what these numerical values represent.
+        """
+        question_vector = np.asarray(embeddings_model.predict(question))
+        gen_question_vectors = []
```

Comment:
Nit: it would be nice to use list comprehensions where possible instead of creating an empty list and appending to it in a for loop.

Reply:
Removed after changing the scoring function.